aten/src/ATen/core/type.cpp

```diff
@@ -773,19 +774,21 @@ VaryingShape<Stride> TensorType::computeStrideProps(
 
   std::vector<Stride> stride_properties;
   for (size_t i = 0; i < stride_indices.size(); i++) {
-    Stride s{stride_indices[i], false, strides[stride_indices[i]]};
-    // innermost stride expected to be 1
-    // TODO: turn contiguous_ into an enum CONTIGUOUS, NONCONTIGUOUS,
-    // BROADCASTED
-    if (i == 0) {
-      s.contiguous_ = strides[stride_indices[i]] == 1;
-    } else {
-      s.contiguous_ = strides[stride_indices[i]] == 1 ||
-          (strides[stride_indices[i]] != 0 &&
-           strides[stride_indices[i]] ==
-               strides[stride_indices[i - 1]] * sizes[stride_indices[i - 1]]);
+    bool contiguous_ = tensor_continuity;
+    if (!contiguous_) {
```

Comment:
Why not use `if(!tensor_continuity)`

Reply:
@kshitij12345 Note later we want to use `contiguous_` to create `stride_properties.emplace_back(stride_indices[i], contiguous_, strides[stride_indices[i]]);`. But we don't really want to reuse `tensor_continuity` since `contiguous_` is per dimension while `tensor_continuity` is for the whole Tensor.